File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;

class Solution {
    public int solution(int[] array) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : array) {
            if (map.containsKey(num)) {
                map.put(num, map.get(num) + 1);
            } else {
                map.put(num, 1);
            }
        }

        int max = -1;
        int idx = 0;
        for (Integer integer : map.keySet()) {
            if (map.containsKey(integer) && (map.get(integer) > max)) {
                    max = map.get(integer);
                    idx = integer;
            }
        }

        int count = 0;
        for (Integer integer : map.keySet()) {
            if (max == map.get(integer)) {
                count++;
            }
        }

        return count > 1 ? -1 : idx;
    }
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    counts = {}
    for num in array:
        counts[num] = counts.get(num, 0) + 1

    max_count = max(counts.values())

    modes = [num for num, count in counts.items() if count == max_count]

    return modes[0] if len(modes) == 1 else -1